src/eterna/mode/Mol3DGate.ts

```diff
@@ -0,0 +1,291 @@
+// kkk Mol3DGate.ts --- manage 3d part of EternaGame
+
+import * as NGL from "ngl";
+import { RNABase } from "eterna/EPars";
+import SecStruct from "eterna/rnatypes/SecStruct";
+import Bitmaps from "eterna/resources/Bitmaps";
+import ThreeView from "./ThreeView";
+import GameMode from "./GameMode";
+
+/*
+test pattern
+((((((((((.....((((((((....(((((((.............))))..)))...)))))).)).((.((....((((((((...))))))))....)).))...)))))))))).
+*/
+const isMobile = () => {
+  const checker = {
+    Android: function Android() {
+      return navigator.userAgent.match(/Android/i);
+    },
+    BlackBerry: function BlackBerry() {
+      return navigator.userAgent.match(/BlackBerry/i);
+    },
+    iOS: function iOS() {
+      return navigator.userAgent.match(/iPhone|iPad|iPod/i);
+    },
+    Opera: function Opera() {
+      return navigator.userAgent.match(/Opera Mini/i);
+    },
+    Windows: function Windows() {
+      return (
+        navigator.userAgent.match(/IEMobile/i) ||
+        navigator.userAgent.match(/WPDesktop/i)
+      );
+    },
+    any: function any() {
+      return (
+        checker.Android() ||
+        checker.BlackBerry() ||
+        checker.iOS() ||
+        checker.Opera() ||
+        checker.Windows()
+      );
+    },
+  };
+  return !!checker.any();
+};
+
+const DectectDevice = () => {
+  isMobile();
+};
+
+interface PixiCifCheckerCallback {
+  (component: NGL.Structure | null): void;
+}
+
+export default class Mol3DGate {
+  public stage: NGL.Stage;
+  public component: NGL.Component | null;
+  protected backboneElement: NGL.RepresentationElement | null;
+  protected baseElement: NGL.RepresentationElement | null;
+  protected ballstickElement: NGL.RepresentationElement | null;
+  public static scope: Mol3DGate;
```

Comment:
I've made comments elsewhere in this review about some specific instances, but it should be possible to avoid making this a singleton by relying on child elements firing events that are passed up to the GameMode.

Reply:
I removed ballstickElement and moved others into GameMode

Reply:
FYI in case it was unclear, my intent of this comment was about `scope`, not the elements specifically